### PR TITLE
Fight Proposals

### PR DESCRIPTION
**Pull request by antonypap**

### CODE REVIEW

File: pkg/infra/teams/team3/agent3.go [hunk 1]
```diff
@@ -69,6 +75,12 @@ func (a *AgentThree) UpdateInternalState(baseAgent agent.BaseAgent, _ *commons.I
 	if enable {
 		a.UpdatePersonality(baseAgent)
 	}
```
Comment (MattSScott) [inline pkg/infra/teams/team3/agent3.go:77]: Better variable name please - operation not clear

File: pkg/infra/teams/team3/election_strat.go [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
-var (
-	w1, w2      float64
-	pastHP      []int
-	pastStamina []int
-)
+var w1 = make(map[commons.ID]float64)
+var w2 = make(map[commons.ID]float64)
+
+var pastHP = make(map[commons.ID]int)
+var pastStamina = make(map[commons.ID]int)
```
Comment (MattSScott) [inline pkg/infra/teams/team3/election_strat.go:22]: Minor change, but convention is this:

`var` is used before initialisation - when the initial value is unknown (such as with `var x float64`)
we use the 'tusk' operator (`:=`) for assigning when initialisation is known

Worth using `w1 := make(map...)`

File: pkg/infra/teams/team3/election_strat.go [hunk 3]
```diff
@@ -140,7 +119,7 @@ func (a *AgentThree) calcW2(baseAgent agent.BaseAgent, w2 float64) float64 {
 	itr := a.fightDecisionsHistory.Iterator()
 	for !itr.Done() {
 		res, _ := itr.Next()
-		i += 1
+		i = i + 1
```
Comment (MattSScott) [inline pkg/infra/teams/team3/election_strat.go:122]: `i++` ?

**Review by BusterBlack — APPROVED** [review]
LGTM for sanctions/messaging

File: pkg/infra/teams/team3/election_strat.go (hunk 2 above)
Comment (MattSScott) [reply pkg/infra/teams/team3/election_strat.go:22]: Also, there is a clash in variable name:

w1 is used here as a map and also in `calcW1` as a `float64` - this makes readability less clear

File: pkg/infra/teams/team3/election_strat.go [hunk 4]
```diff
@@ -179,70 +158,170 @@ func (a *AgentThree) calcW2(baseAgent agent.BaseAgent, w2 float64) float64 {
...
 	needs := 5.0
-	fairness := [][]int{}
-
-	itr := agentState.Iterator()
+	// fairness := make(map[commons.ID]float64)
+
+	// Number of agents to sample for KA (fixed)
+	intendedSample := float64(a.numAgents) * a.sample_percent
+	maxLength := float64(vAS.Len())
+	sampleLength := int(math.Min(intendedSample, maxLength))
```
Comment (MattSScott) [inline pkg/infra/teams/team3/election_strat.go:173]: nice, good change

File: pkg/infra/teams/team3/election_strat.go [hunk 5]
```diff
@@ -179,70 +158,170 @@ func (a *AgentThree) calcW2(baseAgent agent.BaseAgent, w2 float64) float64 {
...
+				w1[id] = 0.0
+				w2[id] = 0.0
+				pastHP[id] = GetStartingHP()
+				pastStamina[id] = GetStartingStamina()
+			}
 
-	for !itr.Done() {
-		id, hiddenState, _ := itr.Next()
-		idN, _ := strconv.Atoi(id)
+			hiddenState, _ := vAS.Get(id)
```
Comment (MattSScott) [inline pkg/infra/teams/team3/election_strat.go:190]: pretty sure you don't need the `_` here, as it's the second return value

File: pkg/infra/teams/team3/election_strat.go (hunk 2 above)
Comment (BusterBlack) [reply pkg/infra/teams/team3/election_strat.go:22]: yes good spot!

**Pull request merged**